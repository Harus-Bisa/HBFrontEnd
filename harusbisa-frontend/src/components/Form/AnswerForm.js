import React from "react";
import {Input, CustomInput} from "reactstrap";
import { setAnswer, setCorrectAnswer } from "../../redux/actions";
import { connect } from "react-redux";

function AnswerForm(props){
    var [correct, setCorrect] = React.useState(false)
    var [answer, setLocalAnswer] = React.useState("");
    
    const changeAnswer = (event) =>{
        setLocalAnswer(event.target.value)
        props.setAnswer(props.option, event.target.value)
    }

    const changeCorrectAnswer = ()=>{
        setCorrect(!correct)
        props.setCorrectAnswer(props.option)
    }
    return(
        <div className="row" style={{marginTop:"15px", marginBottom:'15px'}}>
            <div className="col-1" style={{display:'flex'}}>
                <p style={{margin:'auto'}}>{String.fromCharCode(props.option+65)}.</p>
            </div> 
            <div className="col-11">
                <div className="form-control" style={{height:'auto'}}>
                    <Input
                        type="textArea"
                        placeholder={"Tulis jawaban anda disini"}
                        style={{padding:"0 0 .75rem 0", border:"none", boxShadow:"none"}}
                        value={answer}
                        onChange={changeAnswer}
                    />
                    <div style={{display:'flex', justifyContent:'flex-end'}}>
                        <CustomInput type="switch" id={"switch-"+props.option} value={correct} label="Jawaban benar" onChange={changeCorrectAnswer}/>
                    </div>
                </div>
            </div>   
        </div>
    )
}
export default connect(null, {setAnswer, setCorrectAnswer})(AnswerForm)