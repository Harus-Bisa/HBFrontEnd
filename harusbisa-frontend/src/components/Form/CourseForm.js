import React from "react";
import {connect} from "react-redux";
import { addCourse, editCourse } from "../../redux/actions";
import { Form, FormGroup, Label, Input, FormFeedback, FormText, Col} from 'reactstrap';
import { HB_YELLOW } from "../../css/constants/color";
import { Button } from "@material-ui/core";

function mapStateToProps(state, ownProps){
    if (ownProps.id){
        let courses = state.courses
        for (let i=0; i<courses.length; i++){
            if (courses[i]._id === ownProps.id){
                return{
                    course: courses[i]
                }
            }
        }
        return null
    }
    else{
        return {
            course:{
                course_name: "",
                start_term: "",
                end_term: ""
            }
        }
    }
    
}
function CourseForm(props){
    const [name, setName] = React.useState(props.course.course_name);
    const [startDate, setStartDate] = React.useState(props.course.start_term);
    const [endDate, setEndDate] = React.useState(props.course.end_term);
    
    const submit = (event) =>{
        event.preventDefault();
        if (props.id){
            props.editCourse(props.id, name, startDate, endDate);
        }
        else{
            props.addCourse(name, startDate, endDate);
        }
        props.closePopup();
    }
    return(
        <div className="container-fluid">
            {/* <form onSubmit={submit}>
                <label>Course Name*</label>
                <input type="text" value={name} onChange={(event) => setName(event.target.value)}></input>
                <br/>
                <label>Start Date*</label>
                <input type="text" value={startDate} onChange={(event) => setStartDate(event.target.value)}></input>
                <br/>
                <label>End Date*</label>
                <input type="text" value={endDate} onChange={(event) => setEndDate(event.target.value)}></input>
                <br/>
                <button type="submit">Submit</button>
            </form> */}
            <div className="row" style={{borderBottom:'2px solid '+HB_YELLOW}}>
                <div className="col-12" style={{display:'flex', justifyContent:'center', padding:'1rem 2rem'}}>
                    <h3>Tambahkah Kelas Baru</h3>
                </div>
            </div>
            <div className="content">
                <Form onSubmit={submit}>
                    <FormGroup row>
                        <Label sm={5}>Nama Kelas</Label>
                        <Col sm={7}>
                            <Input type="text" value={name} placeholder={"Contoh: Biologi Molekuler Kelas A"} onChange={(event) => setName(event.target.value)}/>
                            <FormText>Nama ini adalah nama yang akan dilihat siswa Anda.</FormText>
                        </Col>
                    </FormGroup>
                    <FormGroup row>
                        <Label sm={5}>Periode Kelas</Label>
                        <Col sm={7}>
                            <div className="row">
                                <div className="col-5">
                                    <Input type="select" name="select" id="exampleSelect">
                                        <option>Juli 2019</option>
                                        <option>Juli 2020</option>
                                        <option>Juli 2021</option>
                                        <option>Juli 2022</option>
                                        <option>Juli 2023</option>
                                    </Input>
                                </div>
                                <div className="col-2">
                                    <p>-</p>
                                </div>
                                <div className="col-5">
                                    <Input type="select" name="select" id="exampleSelect">
                                        <option>Juli 2019</option>
                                        <option>Juli 2020</option>
                                        <option>Juli 2021</option>
                                        <option>Juli 2022</option>
                                        <option>Juli 2023</option>
                                    </Input>
                                </div>
                            </div>
                        </Col>
                    </FormGroup>
                    <Button className="prof-button" type="submit">Submit</Button>
                </Form>
            </div>
        </div>
    )
}

export default connect(mapStateToProps,{addCourse, editCourse})(CourseForm);