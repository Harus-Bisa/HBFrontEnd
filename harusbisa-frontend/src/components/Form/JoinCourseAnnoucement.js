import React from "react";
import { Button } from "@material-ui/core";
import {connect} from "react-redux";

function JoinCourseAnnoucement(props){
    const style={
        text:{
            marginBottom:"1rem"
        }
    }
    return(
        <div className="container-fluid">
            <div className="content">
                <div className="row">
                    <div className="col-12">
                        <h5 style={style.text}>Mintalah siswa Anda membuat akun mereka sendiri</h5>
                    </div>
                </div>
                <div className="row">
                    <div className="col-12">
                        <p style={style.text}>Salin, kemudia bagikan link ini dengan siswa Anda.</p>
                    </div>
                </div>
                <div className="row" style={{marginBottom:'1rem'}}>
                    <div className="col-9" style={{display:'flex'}}>
                        <div style={{border:'1.5px solid #f4f4f4', padding:'4.5px 8px', margin:'auto 0', width:"100%"}}>
                            <p>{"www.harusbisa.net/join/" + props.joinCode}</p>
                        </div>
                    </div>
                    <div className="col-3" style={{display:'flex'}}>
                        <Button className="prof-button" fullWidth style={{margin:'auto'}}>Salin</Button>
                    </div>
                </div>
                <div className="row" style={{marginBottom:'1rem'}}>
                    <div className="col-12">
                        <p>Atau mintalah siswa Anda mengunjungi www.harusbisa.net/signup dan masukkan kode kelas Anda <strong>{props.joinCode}</strong></p>
                    </div>
                </div>
                <div className="row justify-content-end">
                    <div className="col-3">
                        <Button className="prof-button" fullWidth onClick={props.closePopup}>Selesai</Button>
                    </div>
                </div>
                
            </div>
        </div>
    )
}

function MapStateToProps(state){
    return{
        joinCode:state.courses[state.courses.length-1].join_code
    }
}
export default connect(MapStateToProps)(JoinCourseAnnoucement);