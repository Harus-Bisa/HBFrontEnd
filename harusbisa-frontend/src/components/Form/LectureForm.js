import React from "react";
import {connect} from "react-redux";
import { Form, FormGroup, Label, Input, FormText, Col} from 'reactstrap';
import { Button, Slider } from "@material-ui/core";
import {addLecture, editLecture} from "../../redux/actions";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

function LectureForm(props){
    const type = (props.id && props.lecture ? "EDIT" : "ADD")
    var [date, setDate] = React.useState(type === "ADD" ? new Date() : new Date(props.lecture.date));
    var [lectureDescription, setLectureDescription] = React.useState(type === "ADD" ? "" : props.lecture.lectureDescription)
    var [participationPercentage, setParticipationPercentage] = React.useState(type === "ADD" ? Number(100) : props.lecture.participationRewardPercentage);
    

    const submit = (event) =>{
        event.preventDefault();
        let dateMS = (new Date(date)).getTime()
        if(type === "ADD"){
            props.addLecture(dateMS, lectureDescription, participationPercentage, props.courseId, props.role)
        }
        else if (type === "EDIT"){
            //editLecture
            props.editLecture(dateMS, lectureDescription, participationPercentage, props.courseId, props.id, props.role)
        }
        props.closePopup()
        
    }
    return(
        <div className="container-fluid student-course-form">
            <div className="row">
                <div className="content col-md-5" style={{display:'flex'}}>
                    <h1 style={{margin:'auto', fontSize:"30px"}}>{type === "ADD" ? "Tambah Sesi":"Edit Sesi"}</h1>
                </div>
                <div className="content col-md-7">
                <Form onSubmit={submit}>
                    <FormGroup style={{display:'flex', flexDirection:'column'}}>
                        <Label>Nama Sesi</Label>
                        <DatePicker 
                            id="lectureDate" 
                            selected={date} 
                            onChange={(date) => setDate(date)} 
                            dateFormat="dd/MM/yyyy"
                        />
                        <FormText>Tanggal ini akan menjadi nama sesi anda.</FormText>
                    </FormGroup>
                    <FormGroup row>
                        <Label sm={4}>Deskripsi</Label>
                        <Col sm={8}>
                            <Input
                                type="text"
                                id="lectureDescription"
                                value={lectureDescription}
                                onChange={(event) => setLectureDescription(event.target.value)}
                            />
                        </Col>
                    </FormGroup>
                    <FormGroup>
                        <Label>Persentase</Label>
                        <div className="container">
                            <Slider
                                value={participationPercentage}
                                onChange={(event, newValue) => setParticipationPercentage(newValue)}
                                min={0}
                                max={100}
                            />
                        </div>
                        <div style={{display:'flex', justifyContent:"space-between"}}>
                            <div>
                                <p>Partisipasi: {participationPercentage}%</p>
                            </div>
                            <div>
                                <p>Benar: {100 - participationPercentage}%</p>
                            </div>
                        </div>
                    </FormGroup>
                    <div style={{justifyContent:'flex-end', display:'flex'}}>
                        <Button type="submit" className="prof-button">{type === "ADD" ? "Tambahkan":"Edit"}</Button>
                    </div>
                </Form>
                </div>
            </div>
        </div>
    )
}

function mapStateToProps(state, ownProps){
    for (let i=0; i<state.course.lectures.length; i++){
        let lecture = state.course.lectures[i]
        if (lecture.lectureId === ownProps.id){
            return{
                role: state.role,
                courseId: state.course.courseId,
                lecture: lecture
            }
        }
    }
    return{
        role: state.role,
        courseId: state.course.courseId
    }
    
}
export default connect(mapStateToProps,{addLecture, editLecture})(LectureForm);