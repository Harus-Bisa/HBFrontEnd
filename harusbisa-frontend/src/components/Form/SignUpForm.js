import React from "react";
import {connect} from "react-redux";
import {Form, FormGroup, Label, Col, Input} from "reactstrap";
import {Button} from "@material-ui/core";
import { signup } from "../../redux/actions";
import ErrorMessage from "../Error/ErrorMessage";
import {withRouter} from "react-router-dom";

function SignUpForm(props){
    var [firstName, setFirstName] = React.useState("")
    var [lastName, setLastName] = React.useState("")
    var [email, setEmail] = React.useState("")
    var [school, setSchool] = React.useState("")
    var [role, setRole] = React.useState("")
    var [password, setPassword] = React.useState("")

    var verified = firstName && lastName && email && school && role && password;
    const submit = (event) =>{
        event.preventDefault();
        props.signup(firstName, lastName, email, school, role, password);
        if (!props.error){
            props.history.push("/login")
        }
    }
    return(
        <div className="container">
            {props.error && <ErrorMessage/>}
            <Form onSubmit={submit}>
                <FormGroup row>
                    <Label sm={3}>Nama Depan</Label>
                    <Col sm={9}>
                        <Input type="text" id="firstName" value={firstName} onChange={(event) => setFirstName(event.target.value)}/>
                    </Col>
                </FormGroup>
                <FormGroup row>
                    <Label sm={3}>Nama Belakang</Label>
                    <Col sm={9}>
                        <Input type="text" id="lastName" value={lastName} onChange={(event) => setLastName(event.target.value)}/>
                    </Col>
                </FormGroup>
                <FormGroup row>
                    <Label sm={3}>Email</Label>
                    <Col sm={9}>
                        <Input type="email" id="email" value={email} onChange={(event) => setEmail(event.target.value)}/>
                    </Col>
                </FormGroup>
                <FormGroup row>
                    <Label sm={3}>Perguruan Tinggi</Label>
                    <Col sm={9}>
                        <Input type="text" id="school" value={school} onChange={(event) => setSchool(event.target.value)}/>
                    </Col>
                </FormGroup>
                <FormGroup check>
                    <Label check>
                        <Input type="radio" id="role-faculty" checked={role === "faculty"} onChange={() => setRole("faculty")} />{' '}
                        Dosen
                    </Label>
                </FormGroup>
                <FormGroup check>
                    <Label check>
                        <Input type="radio" id="role-student" checked={role === "student"} onChange={() => setRole("student")}/>{' '}
                        Mahasiswa
                    </Label>
                </FormGroup>
                <FormGroup row>
                    <Label sm={3}>Password</Label>
                    <Col sm={9}>
                        <Input type="password" id="password" value={password} onChange={(event) => setPassword(event.target.value)}/>
                    </Col>
                </FormGroup>
                
                <div className="row justify-content-end">
                    <div className="col-3">
                        <Button fullWidth className="prof-button" type="submit" disabled={!verified}>Sign Up!</Button>
                    </div>
                </div>
            </Form>
        </div>
    )

}

function mapStateToProps(state){
    return{
        error: state.error
    }
}
export default connect(mapStateToProps, {signup})(withRouter(SignUpForm))