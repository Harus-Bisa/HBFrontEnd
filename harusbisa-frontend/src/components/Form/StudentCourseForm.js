import React from "react";
import { Button } from "@material-ui/core";

function StudentCourseForm(props){
    var [joinCode, setJoinCode] = React.useState("")
    const submit = (event) =>{
        event.preventDefault();
        console.log(joinCode)
        props.closePopup()
    }
    return(
        <div className="container-fluid">
            <div className="row">
                <div className="content col-md-6" style={{background:"linear-gradient(41.38deg, #9C27D5 5.56%, #6311AB 112.78%)", display:'flex'}}>
                    <h1 style={{color:"white", margin:'auto'}}>Tambah Kelas</h1>
                </div>
                <div className="content col-md-6">
                    <p style={{marginBottom:'2rem'}}>Temukan kelas Anda untuk semester ini melalui kode bergabung yang dibagikan oleh dosen Anda.</p>
                    <form onSubmit={submit} style={{display:'flex', flexDirection:'column'}}>
                        <label>Kode bergabung</label>
                        <input type='text' value={joinCode} onChange={(event) => {setJoinCode(joinCode = event.target.value)}}/>
                        <Button className="student-button" style={{marginTop:'1rem'}}type="submit">Bergabung</Button>
                    </form>
                </div>
            </div>
        </div>
    )
}

export default StudentCourseForm;