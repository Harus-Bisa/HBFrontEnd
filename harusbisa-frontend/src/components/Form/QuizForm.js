import React from "react"
import { Form, FormGroup, Label, Input, Button as RButton, Col} from 'reactstrap';
import {connect} from "react-redux";
import { Button } from "@material-ui/core";
import AnswerForm from "./AnswerForm";

function QuizForm(props){
    var [question, setQuestion] = React.useState("")
    var [numOfAnswers, setNumOfAnswers] = React.useState(2)
    var [points, setPoints] = React.useState(1)
    var [time, setTime] = React.useState(60)
    
    const submit = (event) =>{
        event.preventDefault();
    }
    const incrementNumOfAns = () =>{
        setNumOfAnswers(numOfAnswers + 1)
    }
    const makeAnswers = () =>{
        let answers = []
        for (let i=0; i<numOfAnswers; i++){
            answers.push(
                <AnswerForm option={i} key={i}/>
            )
        }

        return answers
    }
    var verified = question && points
    return(
        <div className="container">
            <div className="row">
                <div className="col-12" style={{display:'flex', justifyContent:'center', padding:'1rem 2rem'}}>
                    <h3>Tambah Pertanyaan</h3>
                </div>
            </div>
            <div className="content">
                <Form onSubmit={submit} style={{minWidth:'50vw'}}>
                    <FormGroup>
                        <Label>Pertanyaan*</Label>
                        <Input type="textarea" id="question" required value={question} placeholder={"Tulis pertanyaan Anda disini"} onChange={(event) => setQuestion(event.target.value)}/>
                    </FormGroup>
                    <FormGroup>
                        <Label>Jawaban*</Label>
                        {makeAnswers()}
                        <div style={{display:'flex', justifyContent:"flex-end"}}>
                            <RButton className="neutral-button" onClick={incrementNumOfAns}>Tambah Pilihan Jawaban</RButton>
                        </div>
                    </FormGroup>
                    <FormGroup>
                        <Label>Setting Pilihan Ganda</Label>
                        <div className="row justify-content-end">
                            <div className="col-11">
                                <FormGroup row>
                                    <Label id="point" sm={3}>Poin Soal Ini*</Label>
                                    <Col sm={9}>
                                        <Input
                                            type="number"
                                            value={points}
                                            onChange={(event) => setPoints(event.target.value)}
                                            required
                                        />
                                    </Col>
                                </FormGroup>
                                <FormGroup row>
                                    <Label sm={3}>Timer*</Label>
                                    <Col sm={6}>
                                        <Input
                                            type="number"
                                            value={time}
                                            onChange={(event) => setTime(event.target.value)}
                                            required
                                        />
                                    </Col>
                                    <Label sm={3}>Detik</Label>
                                </FormGroup>
                            </div>
                        </div>
                    </FormGroup>
                    
                    <div className="row justify-content-end">
                        <div className="col-3">
                            <Button fullWidth className="prof-button">Kembali</Button>
                        </div>
                        <div className="col-3">
                            <Button fullWidth className="prof-button" type="submit" disabled={!verified}>Tambahkan</Button>
                        </div>
                    </div>
                </Form>
            </div>
        </div>
    )
}

export default connect()(QuizForm);