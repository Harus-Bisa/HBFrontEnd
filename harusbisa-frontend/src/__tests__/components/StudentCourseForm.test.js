import React from "react";
import {shallow} from "enzyme";
import '../../test-config';
import { StudentCourseForm } from "../../components/Form/StudentCourseForm";

describe("StudentCourseForm component", () =>{
    let component;
    let mockStudentAddCourse = jest.fn();
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation((init) => [init, setState])
    let mockPreventDefault = jest.fn();

    beforeEach(() =>{
        component = shallow(<StudentCourseForm studentAddCourse={mockStudentAddCourse}/>)
    })
    afterEach(() =>{
        jest.clearAllMocks()
    })

    it("Matches snapshot", () =>{
        expect(component.html()).toMatchSnapshot();
    })
    it ("change state with form entry", () =>{
        var joinCodeField = component.find("#joinCode");
        expect(joinCodeField).toBeDefined();
        expect(setState).not.toHaveBeenCalled()
        joinCodeField.simulate("change",{target:{value:'123'}})
        expect(setState).toHaveBeenCalledTimes(1)
    })
    it("adds course", () =>{
        var form = component.find("form")
        expect(form).toBeDefined()
        expect(mockStudentAddCourse).not.toHaveBeenCalled()
        expect(mockPreventDefault).not.toHaveBeenCalled()
        form.simulate("submit", {preventDefault:mockPreventDefault})
        expect(mockStudentAddCourse).toHaveBeenCalled()
        expect(mockPreventDefault).toHaveBeenCalled()
    })
})
