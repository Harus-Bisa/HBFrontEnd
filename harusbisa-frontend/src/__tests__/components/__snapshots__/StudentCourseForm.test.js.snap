// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentCourseForm component Matches snapshot 1`] = `"<div class=\\"container-fluid student-course-form\\"><div class=\\"row\\"><div class=\\"content col-md-6\\" style=\\"background:linear-gradient(41.38deg, #9C27D5 5.56%, #6311AB 112.78%);display:flex\\"><h1 style=\\"color:white;margin:auto\\">Tambah Kelas</h1></div><div class=\\"content col-md-6\\"><p style=\\"margin-bottom:2rem\\">Temukan kelas Anda untuk semester ini melalui kode bergabung yang dibagikan oleh dosen Anda.</p><form style=\\"display:flex;flex-direction:column\\"><label>Kode bergabung</label><input type=\\"text\\" id=\\"joinCode\\" value=\\"\\"/><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text student-button\\" tabindex=\\"0\\" type=\\"submit\\" style=\\"margin-top:1rem\\"><span class=\\"MuiButton-label\\">Bergabung</span></button></form></div></div></div>"`;
